guide/src/reference/arbitrary-data-with-serde.md

```diff
@@ -94,23 +92,60 @@ example.field2.push([5, 6]);
-temporary JSON stringification. This approach has both advantages and
-disadvantages.
-
-The primary advantage is smaller code size: going through JSON entrenches code
-to stringify and parse floating point numbers, which is not a small amount of
-code. It also supports more types than JSON does, such as `Map`, `Set`, and
-array buffers.
-
-There are two primary disadvantages. The first is that it is not always
-compatible with the default JSON-based serialization. The second is that it
-performs more calls back and forth between JS and Wasm, which has not been fully
-optimized in all engines, meaning it can sometimes be a speed
-regression. However, in other cases, it is a speed up over the JSON-based
-stringification, so &mdash; as always &mdash; make sure to profile your own use
-cases as necessary.
+## An alternative approach - using JSON
+
+`serde-wasm-bindgen` works by directly manipulating JavaScript values. This
+requires a lot of calls back and forth between Rust and JavaScript, which can
+sometimes be slow. An alternative way of doing this is to serialize values to
+JSON, and then parse them on the other end. Browsers' JSON implementations are
+usually quite fast, and so this approach can outstrip `serde-wasm-bindgen`'s
+performance in some cases.
+
+That's not to say that using JSON is always faster, though - the JSON approach
+can be anywhere from 2x to 0.2x the speed of `serde-wasm-bindgen`, depending on
+the JS runtime and the values being passed. It also leads to larger code size
+than `serde-wasm-bindgen`. So, make sure to profile each for your own use
+cases.
+
+This approach is implemented in [`gloo_utils::format::JsValueSerdeExt`]:
+
+```toml
+# Cargo.toml
+[dependencies]
+gloo-utils = { version = "0.1", features = ["serde"] }
+```
+
+```rust
+use gloo_utils::format::JsValueSerdeExt;
+
+#[wasm_bindgen]
+pub fn send_example_to_js() -> JsValue {
+    let mut field1 = HashMap::new();
+    field1.insert(0, String::from("ex"));
+    let example = Example {
+        field1,
+        field2: vec![vec![1., 2.], vec![3., 4.]],
+        field3: [1., 2., 3., 4.]
+    };
+
+    JsValue::from_serde(&example).unwrap()
+}
+
+#[wasm_bindgen]
+pub fn receive_example_from_js(val: JsValue) {
+    let example: Example = val.into_serde().unwrap();
+    ...
+}
+```
+
+[`gloo_utils::format::JsValueSerdeExt`]: https://docs.rs/gloo-utils/latest/gloo_utils/format/trait.JsValueSerdeExt.html
+
+## `JsValue::from_serde` / `JsValue::into_serde`
+
+In previous versions of `wasm-bindgen`, `JsValue::from_serde` and
+`JsValue::into_serde` were the recommended way of using Serde to convert
+to/from JS values. These were implemented using the JSON approach with
+`serde_json`. This caused problems when certain features of `serde_json` and
+other crates were enabled that caused it to depend on `wasm-bindgen`, creating
+a circular dependency, which is illegal in Rust and caused people's code to
+fail to compile. So, they were deprecated.
```

Comment:
It didn't end up being deprecation (it's not saying “don't use this anymore, we may remove support for it in some upcoming version”), so maybe
```suggestion
fail to compile. So, they were moved out of `wasm-bindgen` into `gloo_utils` to
resolve the circular dependency and replaced by `serde-wasm-bindgen` as the
default approach to use.
```
... or something like that.